inference_segmentation.py

```diff
@@ -242,101 +243,140 @@ def classifier(params, img_list, model, device, working_folder):
                    delimiter=',')
 
 
-def main(params: dict):
+def main(params: dict) -> None:
     """
-    Identify the class to which each image belongs.
-    :param params: (dict) Parameters found in the yaml config file.
+    Function to manage details about the inference on segmentation task.
 
+    1. Read the parameters from the config given.
+    2. Read and load the state dict from the previous training or the given one.
+    3. Make the inference on the data specifies in the config.
+
+    -------
+    :param params: (dict) Parameters found in the yaml config file.
     """
-    since = time.time()
+    # since = time.time()
+
+    # PARAMETERS
+    mode = get_key_def('mode', params, expected_type=str)
+    task = get_key_def('task_name', params['task'], expected_type=str)
+    model_name = get_key_def('model_name', params['model'], expected_type=str).lower()
+    num_classes = len(get_key_def('classes_dict', params['dataset']).keys())
+    modalities = read_modalities(get_key_def('modalities', params['dataset'], expected_type=str))
+    BGR_to_RGB = get_key_def('BGR_to_RGB', params['dataset'], expected_type=bool)
+    num_bands = len(modalities)
+    debug = get_key_def('debug', params, default=False, expected_type=bool)
+    # SETTING OUTPUT DIRECTORY
+    try:
+        state_dict = Path(params['inference']['state_dict_path']).resolve(strict=True)
```

Comment:
The state_dict was previously expected to be a path directly pointing to the checkpoint.pth.tar. Your version of state_dict expects a directory, where the last modified checkpoint.pth.tar will be loaded ([line 328](https://github.com/NRCan/geo-deep-learning/pull/208/files#diff-b85c0e821d4ba746c7ae4190dbc4ebafd445d4231a8a1f9796cd238fd79cc052R328)). What is the motivation behind this change? I liked the idea of pointing directly to the file containing weight for the model to use. I like to know exactly which set of weights will be used.